clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,485 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//Package network captures network performance metrics
+//for protocol TCP,UDP,HTTP etc. The metrics are collected for baseline (1:1),
+//scale (N:M) pod ratios.Client and server pods located on different worker
+//nodes exchange traffic for specified time to measure the performance metrics.
+package network
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/api/meta"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework"
+
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework/client"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	workerPodDeploymentManifestFilePath = "$GOPATH/src/k8s.io/perf-tests/clusterloader2/pkg/measurement/common/network/manifests/*deployment.yaml"
+
+	// podReadyTimeOutFactor is used in wait.poll() and is the timeout value.
+	// We assume 4 pods will be created per second.
+	podReadyTimeOutFactor = 0.25
+
+	// podReadyPollInterval is used in wait.poll(). Every 2seconds,check whether pods are ready.
+	podReadyPollInterval = 2
+
+	// initialDelay in seconds after which pod starts sending traffic.
+	// Delay is used to synchronize all client pods to send traffic at same time.
+	initialDelayForTestExecution = 5
+
+	// networkPerformanceMetricsName indicates the measurement name
+	networkPerformanceMetricsName = "NetworkPerformanceMetrics"
+
+	netperfNamespace = "netperf"
+)
+
+const (
+	networkTestRequestFilePath = "$GOPATH/src/k8s.io/perf-tests/clusterloader2/pkg/measurement/common/network/manifests/networktestrequests.yaml"
+
+	customResourceGroup = "clusterloader.io"
+
```

Comment:
let's not add those artificial empty lines

Reply:
Yes , modified.